src/ResourceManager/Network/Commands.Network/ApplicationGateway/RedirectConfiguration/AddAzureApplicationGatewayRedirectConfigurationCommand.cs

```diff
@@ -0,0 +1,49 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Network.Models;
+using System;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.Add, "AzureRmApplicationGatewayRedirectConfiguration"), OutputType(typeof(PSApplicationGateway))]
+    public class AddAzureApplicationGatewayRedirectConfigurationCommand : AzureApplicationGatewayRedirectConfigurationBase
```

Comment:
@akshaysngupta I'm not sure this cmdlet is needed; it is just editing the property of the object passed to it. A user could do something similar to the following:

```powershell
# Create the redirect configuration
$RedirectConfiguration = New-AzureRmApplicationGatewayRedirectConfiguration

# Add the redirect configuration
$ApplicationGateway.RedirectConfigurations.Add($RedirectConfiguration)
```

The only difference between the above two lines and this cmdlet is that the lines above don't check for the existing redirect configuration, but that's an additional line users can use to check.